Python Script/csv_processing.py

```diff
@@ -0,0 +1,457 @@
+            except ValueError:
+                # If the element cannot be converted to a float, add it to the output row as is
+                modified_row.append(element)
+
+        height = float(row[header.index('Z')-1])
+        modified_row.append(get_height_class_code(height))
+        modified_row.append(get_tree_health(modified_row))
+
+        # Write the modified row to the output file
+        csv_writer.writerow(modified_row)
+
+# Read the CSV file and compute the required values
+height_class_codes = []
+heights = []
+areas = []
+tree_colors = []
+tree_segments_count = 0
+forest_coverage_area = 0
+filename = 'crown_metrics_output.csv'
+file_data = pd.read_csv(filename)
+
+with open(filename, mode="r") as csvfile:
+    csv_reader = csv.reader(csvfile)
+    header = next(csv_reader)
+
+    for row in csv_reader:
+        height = float(row[header.index('Z')])
+        area = float(row[header.index('convhull_area')])
+        r = round(float(row[header.index('R')]))
+        g = round(float(row[header.index('G')]))
+        b = round(float(row[header.index('B')]))
+
+        tree_colors.append((r, g, b))
+        heights.append(height)
+        areas.append(area)
+        height_class_codes.append(int(row[header.index('Height Class Code')]))
+        tree_segments_count += 1
+        forest_coverage_area += float(row[header.index('convhull_area')])
+
+forest_coverage_area = round(forest_coverage_area, 6)
+
+# Generate the histogram data
+hist_data, bin_edges, _ = plt.hist(
+    height_class_codes, bins=range(1, 10), align='left', rwidth=0.8)
+
+# Add count labels on top of the bars
+for i in range(len(hist_data)):
+    count = hist_data[i]
+    x_position = bin_edges[i]  # Centering the label on the bar
+    y_position = count + 0.2  # Slightly above the bar
+    plt.text(x_position, y_position, str(int(count)), ha='center', va='bottom')
+
+plt.xlabel("Height Class Code")
+plt.ylabel("Count")
+plt.title("Distribution of Height Class Codes")
+plt.xticks(range(1, 9))
+
+buf = BytesIO()
+plt.savefig(buf, format='png', dpi=300)
+plt.clf()
+buf.seek(0)
+height_class_codes_image = base64.b64encode(buf.read()).decode("utf-8")
+buf.close()
+
+# Generate Height Distribution Plot
+plt.hist(heights, bins=20)
+plt.xlabel('Height (Z)')
+plt.ylabel('Frequency')
+plt.title('Height Distribution')
+
+buf_height = BytesIO()
+plt.savefig(buf_height, format='png', dpi=300)
+plt.clf()
+buf_height.seek(0)
+height_distribution_image = base64.b64encode(buf_height.read()).decode("utf-8")
+buf_height.close()
+
+# Generate Area Distribution Plot
+plt.hist(areas, bins=20)
+plt.xlabel('Convex Hull Area')
+plt.ylabel('Frequency')
+plt.title('Area Distribution')
+
+buf_area = BytesIO()
+plt.savefig(buf_area, format='png', dpi=300)
+plt.clf()
+buf_area.seek(0)
+area_distribution_image = base64.b64encode(buf_area.read()).decode("utf-8")
+buf_area.close()
+
+# Color distribution
+r_values, g_values, b_values = zip(*tree_colors)
```

Comment:
@KulrajbirS  having errors at this line 
<img width="1374" alt="Screenshot 2023-04-22 at 8 16 23 PM" src="https://user-images.githubusercontent.com/34647570/233817813-a8decd26-d813-43e2-b998-8920cc3deb2a.png">
<img width="1037" alt="Screenshot 2023-04-22 at 8 15 43 PM" src="https://user-images.githubusercontent.com/34647570/233817816-c02ef96c-4218-42e4-8d41-f03b2df7743c.png">